src/paas_charm/charm_state.py

```diff
@@ -170,67 +118,68 @@ def from_charm(  # pylint: disable=too-many-arguments,too-many-locals
-                    integration_requirers.s3.to_relation_data()
-                    if integration_requirers.s3
-                    else None
-                ),
-                saml_relation_data=(
-                    integration_requirers.saml.to_relation_data()
-                    if integration_requirers.saml
+                openfga=(
+                    store_info
+                    if (
+                        integration_requirers.openfga
+                        and (store_info := integration_requirers.openfga.get_store_info())
+                    )
                     else None
                 ),
-                rabbitmq_relation_data=(
+                rabbitmq=(
                     integration_requirers.rabbitmq.get_relation_data()
                     if integration_requirers.rabbitmq
                     else None
                 ),
-                tempo_relation_data=(
-                    integration_requirers.tracing.to_relation_data()
-                    if integration_requirers.tracing
+                redis=(
+                    integration_requirers.redis.to_relation_data()
+                    if integration_requirers.redis
+                    else None
+                ),
+                s3=(
+                    integration_requirers.s3.to_relation_data()
+                    if integration_requirers.s3
                     else None
                 ),
-                smtp_relation_data=(
+                saml=(
+                    integration_requirers.saml.to_relation_data()
+                    if integration_requirers.saml
+                    else None
+                ),
+                smtp=(
                     smtp_data
                     if (
                         integration_requirers.smtp
                         and (smtp_data := integration_requirers.smtp.get_relation_data())
                     )
                     else None
                 ),
-                openfga_relation_data=(
-                    store_info
-                    if (
-                        integration_requirers.openfga
-                        and (store_info := integration_requirers.openfga.get_store_info())
-                    )
+                tracing=(
+                    integration_requirers.tracing.to_relation_data()
+                    if integration_requirers.tracing
                     else None
                 ),
             )
-            peer_fqdns = None
-            if secret_storage.is_initialized and (
-                peer_unit_fqdns := secret_storage.get_peer_unit_fdqns()
-            ):
-                peer_fqdns = ",".join(peer_unit_fqdns)
-        except InvalidS3RelationDataError as exc:
-            raise CharmConfigInvalidError("Invalid S3 relation data") from exc
-        except InvalidSAMLRelationDataError as exc:
-            raise CharmConfigInvalidError("Invalid SAML relation data") from exc
+        except InvalidRelationDataError as exc:
```

Comment:
All `RelationDataError` within our custom integration wrappers inherit from `InvalidRelationDataError`. This is to prevent importing "un-fetched" libraries and it's `InvalidXXRelationDataError`s.
That is to say:
- if redis integration lib was not fetched
- we use `InvalidRedisRelationDataError`
- NameError is raised

Dynamically importing such errors based on the presence of the integration is an overkill.